### PR TITLE
mesa: update to 22.0.4

### PR DESCRIPTION
**Pull request by zlice**
#### Testing the changes
- I tested the changes in this PR: **YES**

Performance seems better? May be placebo.

#### Local build testing
- I built this PR locally for my native architecture, x86_64

#### Comments

Tested this on Intel i7-6700k iGPU and AMD 5700. It looks like older cards are being dropped and the build complains about some "amber" branch? Project "amber"? I don't know what that is in Mesa land. Looks like i915/i965 are dropped but my 6700k said it was using 'iris' graphics. [Gentoo's wiki](https://wiki.gentoo.org/wiki/Intel) has a table that shows Haswell (Intel 4xxx CPUs) are the last to not support iris driver.

I'm sure there are people who know way more than me about Mesa's changes. Just figured I'd put this here since it seems to be working.

### CODE REVIEW

**Comment by tsndqst** [comment]
I believe the `dri` i915 driver was dropped but the `gallium` driver is still there.
https://gitlab.freedesktop.org/mesa/mesa/-/blob/mesa-22.0.2/meson_options.txt#L77

I'm not sure about the i965 driver though.  Oh, it looks like that was removed with https://gitlab.freedesktop.org/mesa/mesa/-/commit/cdde031ac2c8124721655532ee6f4149e20e9c61
This same commit added i915 to the list of gallium drivers.

File: srcpkgs/mesa/template
Comment (tsndqst) [inline srcpkgs/mesa/template:94]:
```suggestion
	_gallium_drivers+=",crocus,iris,i915"
```
I suggest adding i915 to the gallium drivers list.  I cross-compiled this for i686 and tested on my EeePC.
`glxinfo -B` in X shows this output:
```name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Mesa Project (0x8086)
    Device: i915 (chipset: 945GME) (0x27ae)
    Version: 22.0.2
    Accelerated: yes
    Video memory: 192MB
    Unified memory: yes
    Preferred profile: compat (0x2)
    Max core profile version: 0.0
    Max compat profile version: 2.1
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 2.0
OpenGL vendor string: Mesa Project
OpenGL renderer string: i915 (chipset: 945GME)
OpenGL version string: 2.1 Mesa 22.0.2
OpenGL shading language version string: 1.20

OpenGL ES profile version string: OpenGL ES 2.0 Mesa 22.0.2
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 1.0.16
```

I did notice that `sway` (wayland) uses the `softpipe` renderer.

**Comment by wundrweapon** [comment]
I'm getting a build error when trying to build elfutils? I'd also like to mention [22.0.3 is already out](https://docs.mesa3d.org/relnotes/22.0.3.html) with some bug fixes.

I'm hoping to see 22.x.x on Void soon; I just got a 6000 series card and I'm dying to see what it can do

**Comment by zlice** [comment]
Not sure what's going on with your elfutils, I haven't had to touch it. Could try 22.0.3, didn't notice anything weird with .2 yet but why not.

**Comment by wundrweapon** [comment]
Rebasing fixed the build -- oops! 22.0.3 builds just fine, too, but unfortunately is not enough to get ray tracing in DOOM Eternal. I note that the game reports Vulkan 1.3.x but the `vulkan-loader` and `Vulkan-Headers` packages are 1.2.x, but that should be no issue because Mesa officially implements 1.2.x

**Comment by HadetTheUndying** [comment]
This looks fine, but I think you need to squash it into one commit. A Maintainer would need to come in and verify, but that's how it's been in the past when I've updated the package. I'll start testing 22.0.3 tomorrow. It built fine on all my my cross-arch builds.

File: srcpkgs/mesa/template
Comment (tsndqst) [reply srcpkgs/mesa/template:93]: @zlice it looks like you lost the change that added i915 to the list of gallium drivers.

**Review by tsndqst — CHANGES_REQUESTED** [review]

**Comment by zlice** [comment]
Removed 'wip'. If I should make another branch since this one is typo'd and incorrect version, or make any other updates to the template and drivers, lmk.

**Comment by HadetTheUndying** [comment]
Is anything holding this PR up? 22.0.4 is now out

**Comment by zlice** [comment]
Anyone else using 22.0.4 ? I just went back to 22.0.3 to see if these weird glitches I've been having are from that or not. Just had a desktop lockup for e.g. (visible but not responding).

**Comment by RicArch97** [comment]
Runs fine for me on Sway. Tested in some games and I'm noticing some nice performance improvements with my 6900XT card on Mesa 22.

**Comment by sug0** [comment]
I tested this patch on my machine, it's also running fine.

File: srcpkgs/mesa/patches/add-use-elf-tls.patch
Comment (ericonr) [inline srcpkgs/mesa/patches/add-use-elf-tls.patch:2]: Please keep this header.

File: srcpkgs/mesa/patches/megadriver-symlinks.patch
Comment (zlice) [reply srcpkgs/mesa/patches/megadriver-symlinks.patch:12]: iirc it was added, will double check

**Comment by RicArch97** [comment]
Mesa 22.1.1 was released recently so it might be preferable to update to that instead. I'm already running it on my x86_64 glibc machine, haven't noticed issues so far.

**Comment by zlice** [comment]
another update? xp we'll be chasing versions forever at this point

**Comment by sug0** [comment]
if the current version in the PR works without issues might as well merge it for the time being, IMO

**Comment by RicArch97** [comment]
> another update? xp we'll be chasing versions forever at this point

We are, and I suggest we keep it in check as graphics drivers are quite important packages in my opinion, on my Radeon 6000 GPU every feature release brings quite significant improvements

**Comment by zlice** [comment]
Was there something that replaced this?

**Pull request closed without merging**
